portal/patient/_machine_config.php

```diff
@@ -85,3 +86,36 @@
 if (GlobalConfig::$CONNECTION_SETTING->Multibyte && !function_exists('mb_strlen')) {
     die('<html>Multibyte extensions are not installed but Multibyte is set to true in _machine_config.php</html>');
 }
+
+function resolveHost(): string
+{
+    if (!empty($GLOBALS['site_addr_oath'])) {
+        $host = rtrim(trim($GLOBALS['site_addr_oath']), "/");
+        return rtrim(trim($host . $GLOBALS['webroot']), "/");
+    }
+    $scheme = $_SERVER['REQUEST_SCHEME'] . "://";
+    $possibleHostSources = array('HTTP_X_FORWARDED_HOST', 'HTTP_HOST', 'SERVER_NAME', 'SERVER_ADDR');
+    $sourceTransformations = array(
+        "HTTP_X_FORWARDED_HOST" => function ($value) {
+            $elements = explode(',', $value);
+            return trim(end($elements));
+        }
+    );
+    $host = '';
+    foreach ($possibleHostSources as $source) {
+        if (!empty($host)) {
+            break;
+        }
+        if (empty($_SERVER[$source])) {
+            continue;
+        }
+        $host = $_SERVER[$source];
+        if (array_key_exists($source, $sourceTransformations)) {
+            $host = $sourceTransformations[$source]($host);
+        }
+    }
+    // remove port
+    $host = preg_replace('/:\d+$/', '', trim($host));
+
+    return rtrim(trim($scheme . $host . $GLOBALS['webroot']), "/");
```

Comment:
btw, can this stuff also capture the port (if using a port)?

Reply:
Yes and no! Not standard ports but others yes. I remove port because I didn't want in curl.